src/code/sys_initial_check.c

```diff
@@ -1,17 +1,114 @@
+/**
+ * File: sys_initial_check.c
+ * Description: Functions for checking for the presence of the Expansion Pak and the correct TV type (PAL/NTSC/MPAL),
+ * and functions for printing error messages directly to screen if not found or incorrect.
+ *
+ * These checks are some of the first functions run in Main, before even setting up the system heap, and any image files
+ * are DMA'd directly to fixed RAM addresses.
+ */
 #include "global.h"
+#include "misc/locerrmsg/locerrmsg.h"
+#include "misc/memerrmsg/memerrmsg.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_initial_check/Check_WriteRGBA16Pixel.s")
+// Address with enough room after to load either of the error message image files before the fault screen buffer at the
+// end of RDRAM
+#define CHECK_ERRMSG_STATIC_SEGMENT \
+    (u8*)(FAULT_FB_ADDRESS - sizeof(gExpansionPakNotInstalledErrorTex) - sizeof(gSeeInstructionBookletErrorTex))
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_initial_check/Check_WriteI4Pixel.s")
+void Check_WriteRGBA16Pixel(u16* buffer, u32 x, u32 y, u32 value) {
+    if (value & 1) {
```

Comment:
This checks if the alpha value of the pixel is not zero. I sort of feel like it might be good to wrap this in an explicit `!= 0`, since it's not exactly a "flag" necessarily. But of course, it doesn't really matter (and runs the risk of being haphazardly reverted in cleanup passes) so I don't feel too strongly on it.
```suggestion
    if ((value & 1) != 0) {
```

Reply:
I'll add a note. Or we could have something like a define for `RGBA16_OPAQUE`?